core/src/Kokkos_HIP_Space.hpp

```diff
@@ -67,6 +67,13 @@
 /*--------------------------------------------------------------------------*/
 
 namespace Kokkos {
+namespace Impl {
+
+template <typename T>
+struct is_hip_type_space : public std::false_type {};
+
+}  // namespace Impl
```

Comment:
I'd prefer if you moved the primary template and its specializations closer to their point of use, that is right before the DeepCopy specializations

Reply:
For me, it makes more sense to define with the memory spaces since the name is not specific to "deep copy" but simply asks if the memory space is of a certain backend type. Thus, it could be easily used for purposes other than selecting deep copy implementations.